invenio_communities/assets/semantic-ui/js/invenio_communities/members/invitations/invitationsModal/InvitationsMembersModal.js

```diff
@@ -0,0 +1,277 @@
+import { ErrorMessage } from "../../components/ErrorMessage";
+import { SelectedMembers } from "./SelectedMembers";
+import { MembersSearchBar } from "./MemberSearchBar";
+import ClassicEditor from "@ckeditor/ckeditor5-build-classic";
+import CKEditor from "@ckeditor/ckeditor5-react";
+
+export class InvitationsMembersModal extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      open: false,
+      selectedMembers: [],
+      role: undefined,
+      message: "",
+      error: undefined,
+      isLoading: false,
+      suggestions: [],
+    };
+  }
+
+  serializePeople = (people) => {
+    return people.map((person) => {
+      return {
+        text: person.profile.full_name,
+        value: person.id,
+        key: person.id,
+        content: (
+          <List divided relaxed>
+            <List.Item>
+              <Image size="mini" src={person.links.avatar} avatar />
+              <List.Content>
+                <List.Header as="a">{person.profile.full_name}</List.Header>
+                <List.Description>
+                  {person.profile.affiliations}
+                </List.Description>
+              </List.Content>
+            </List.Item>
+          </List>
+        ),
+      };
+    });
+  };
+
+  inviteMembers = async () => {
+    const { communityID, updateQueryState, currentQueryState } = this.props;
+    const { selectedMembers, role, message } = this.state;
+    const client = new CommunityInvitationsApi();
+    const serializedMembers = selectedMembers.map((member) => {
+      return { type: "user", id: member.id };
+    });
+    const payload = {
+      members: serializedMembers,
+      role: role,
+      message: message,
+    };
+    try {
+      this.setState({ isLoading: true });
+      await client.createInvite(communityID, payload);
+      await new Promise((resolve) => setTimeout(resolve, 2000));
+      updateQueryState(currentQueryState);
+      this.setState({
+        open: false,
+        isLoading: false,
+        selectedMembers: [],
+        role: undefined,
+        message: "",
+        suggestions: [],
+        error: undefined,
+      });
+    } catch (e) {
+      this.setState({ error: e.response.data, isLoading: false });
+    }
+  };
+
+  roleOptions = () => {
+    const { roles } = this.props;
+    const { role } = this.state;
+
+    return Object.entries(roles).map(([key, value]) => (
+      <Item>
+        <Item.Content>
+          <Item.Header>
+            <Form.Field>
+              <Checkbox
+                radio
+                onClick={() => this.setState({ role: key })}
+                label={value.title}
+                name={value.title}
+                value={key}
+                checked={role === key}
+              />
+            </Form.Field>
+          </Item.Header>
+          <Item.Meta className="ml-25 mt-0">{value.description}</Item.Meta>
+        </Item.Content>
+      </Item>
+    ));
+  };
+
+  getPanes = () => {
```

Comment:
could you create separate components instead of the methods for these panes? it will improve readability and will separate the components' concerns